ivre/db/mongo.py

```diff
@@ -1127,6 +1127,24 @@ def searchports(self, ports, protocol='tcp', state='open', neg=False):
             for port in ports]}}
 
     @staticmethod
+    def searchobjectid(oid, neg=False):
+        """Filters records by their ObjectID.
+        `oid` can be a single or many (as a list or any iterable) object ID(s),
+        specified as strings or an `ObjectID`s."""
+        if isinstance(oid, basestring) or isinstance(oid, bson.objectid.ObjectId):
+            oid = [bson.objectid.ObjectId(oid)]
+        else:
+            oid = [bson.objectid.ObjectId(elt) for elt in oid]
+        if len(oid) == 1:
+            return {'_id': {'$ne': oid[0]} if neg else oid[0]}
+        if neg:
+            return {'$and': [{'_id': {'$ne': bson.objectid.ObjectId(elt)}}
```

Comment:
Avoid, when possible, `$and` and `$or` operators since they make it harder for MongoDB to use indexes. Plus when possible it often makes easier-to-read code.

This part of the code (the `if neg:` and the `else:` blocks) should be rewritten as:
```Python
return {'_id': {'$nin' if neg else '$in': oid}}
```